File: app/src/lib/request/battle.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

export interface SimulateBattleRequest {
  readonly world: WorldId;
  readonly attacker: readonly Squad[];
  readonly defender: readonly Squad[];
  readonly luck: Luck;
  readonly wall: BuildingLevel;
}

File: crates/nil-payload/src/battle.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::battle::Luck;
use nil_core::infrastructure::building::BuildingLevel;
use nil_core::military::squad::Squad;
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SimulateBattleRequest {
  pub world: WorldId,
  pub attacker: Vec<Squad>,
  pub defender: Vec<Squad>,
  pub luck: Luck,
  pub wall: BuildingLevel,
}
